storyCite.lua

```diff
@@ -0,0 +1,248 @@
+					if i == #current-1 then
+						data[key] = data[key] .. " and "
+					elseif i == #current then
+						data[key] = data[key]
+					else
+						data[key] = data[key] .. ", "
+					end
+				end
+			end
+		end
+		
+		--same concept as above but doesn't add links
+		data["issues"] = ""
+		if args["publication"] then
+			local current = h.splitString(args["publication"], args["separator"])
+			for i = 1, #current do
+				data["issues"] = data["issues"] .. current[i]
+				if i == #current-1 then
+					data["issues"] = data["issues"] .. " and "
+				elseif i == #current then
+					data["issues"] = data["issues"]
+				else
+					data["issues"] = data["issues"] .. ", "
+				end
+			end
+		end
+		
+		-- making series text
+		if args["citation series"] then
+			data["series"] = args["citation series"]
+		elseif args["range"] then
+			data["series"] = "\'\'[[" .. util_link.stripPipe(args["range"]) .. "|" .. util_link.stripPipe(util_link.stripDab(args["range"])) .. "]]\'\'"
+		elseif args["series"] then
+			if args["series"] == "[[Doctor Who television stories|''Doctor Who'' television stories]]" then
+				data["series"] = "\'\'[[Doctor Who]]\'\'"
+			else
+				data["series"] = args["series"]
+			end
+			if args["season number"] then
+				data["series"] = data["series"] .. " " .. "[[" .. util_link.stripPipe(args["season number"]) .. "|" .. util_link.stripPipe(util_link.stripDab(args["season number"])) .. "]]"
+			end
+		else
+			data["series"] = ""
+		end
+		
+		-- making release year text
+		if args["release date"] or args["broadcast date"] or args["premiere"] or args["beta release date"] or args["cover date"] then
+		args["release date"] = h.splitString(util_link.stripDab(args["release date"] or args["broadcast date"] or args["premiere"] or args["beta release date"] or args["cover date"]), "-")
+			local releaseYear
+			local releaseEndYear
+			if #args["release date"] >= 2 then
+				releaseYear = args["release date"][1]:gsub("%s+", ""):sub(-4)
+				releaseEndYear = args["release date"][2]:gsub("%s+", ""):sub(-4)
+				if tonumber(releaseEndYear) == nil then
+					releaseEndYear = nil
+				end
+				if tonumber(releaseYear) == nil then
+					releaseYear = releaseEndYear
+					releaseEndYear = nil
+				end
+			else
+				releaseYear = args["release date"][1]:gsub("%s+", ""):sub(-4)
+			end
+			if releaseEndYear and releaseYear and (releaseEndYear ~= releaseYear) then
+				if (tonumber(releaseEndYear) - tonumber(releaseYear)) == 1 then
+					data["release year"] = "in"
+				else
+					data["release year"] = "between"
+				end
+				data["release year"] = data["release year"] .. " the years [[" .. releaseYear .. " (releases)|" .. releaseYear .."]] and [[" .. releaseEndYear .. " (releases)|" .. releaseEndYear .. "]]"
+			else
+				data["release year"] = "in the year [[" .. releaseYear .. " (releases)|" .. releaseYear .."]]"
+			end
+		else
+			data["release year"] = ""
+		end
+		
+		-- making outputText
```

Comment:
move all of this below the smw store stuff and then `return h.makeOutput(data, args)`